src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1235,11 +1235,12 @@ private void OnFileChanged(object sender, FileSystemEventArgs e)
                 }
 
                 TraceFileChangeRestart(e.ChangeType.ToString(), e.FullPath, shutdown);
-                ScheduleRestart(shutdown);
+                ScheduleRestart(shutdown).ContinueWith(t => TraceWriter.Error($"Error restarting host (full shutdown: {shutdown})", t.Exception),
+                    TaskContinuationOptions.ExecuteSynchronously | TaskContinuationOptions.OnlyOnFaulted);
             }
         }
 
-        private void ScheduleRestart(bool shutdown)
+        private async Task ScheduleRestart(bool shutdown)
```

Comment:
Now that we're awaiting this completion, I wonder if this is still a "scheduled" restart?